mocha.el

```diff
@@ -70,11 +71,15 @@ From http://benhollis.net/blog/2015/12/20/nodejs-stack-traces-in-emacs-compilati
 
 (defun mocha-find-project-root ()
   "Find the root of the project."
-  (let ((root-files '(".git" ".hg" ".svn" "package.json")) (dir nil) (i 0))
-    (while (not dir)
-      (setq dir (locate-dominating-file default-directory (nth i root-files)))
-      (setq i (+ i 1)))
-    dir))
+  (f--traverse-upwards (f-exists? (f-expand "package.json" it))))
+
+(defun mocha-command ()
+  "Return Mocha command.
+
+If a locally installed binary exists, use that, otherwise return
+`mocha-command'."
+  (let ((mocha-bin (f-join (mocha-find-project-root) "node_modules" ".bin" "mocha")))
```

Comment:
I'm a little hesitant to make the configured path secondary to the automatic option. I would rather make the default configuration nil, and have it take precedence. 

While it would be unusual for someone to want to use a different mocha than they have locally installed, it would be impossible for them to do it with this scheme.